embedded-service/src/ipc/deferred.rs

```diff
@@ -0,0 +1,98 @@
+//! Definitions for deferred execution of commands
+use core::{
+    marker::PhantomData,
+    sync::atomic::{AtomicU32, Ordering},
+};
+
+use embassy_sync::{blocking_mutex::raw::RawMutex, mutex::Mutex, signal::Signal};
+
+use crate::debug;
+
+/// A unique identifier for a particular command invocation
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct InvocationId(u32);
+
+/// A simple channel for executing deferred commands
```

Comment:
nit: a blank line ensures the top-level decription is brief and sufficiently informative.
```suggestion
/// A simple channel for executing deferred commands.
///
```

Reply:
Done.